src/main/java/frc/robot/Robot.java

```diff
@@ -28,7 +33,9 @@
   private final SendableChooser<String> m_chooser = new SendableChooser<>();
 
   private Drive drivetrain;
-  private LogitechDualAction DriverPad;
+  private LogitechDualAction driverPad;
+  private BallPath ballSubsystem;
+  private Climber climberSubsystem;
```

Comment:
The BallPath and Climber have been declared here but they are never initialized anywhere, so this is a NullPointerException waiting to happen by accident. I would either comment these lines out for now, or in robotSetup() add initializations for these two fields by assigning ex. `new ClimberImpl()`. You will also need to make ClimberImpl implement Climber, which means adding implementations of all the declared methods - they can just stay empty for the time being. This may not *do* anything for now, but at least if you try to bind a gamepad button to some Climber interface method, you get nothing happening rather than an uncaught exception that crashes the robot code.

Reply:
Added implementations of declared methods in latest commit.